src/lib/dataEntities/networks.ts

```diff
@@ -329,9 +349,100 @@ export const l2Networks: L2Networks = {
   },
 }
 
-const getNetwork = async (
+const isParentChain = (chain: Chain): chain is ParentChain => {
+  return chain && 'partnerChainIDs' in chain
+}
+const isChildChain = (chain: Chain): chain is ChildChain => {
+  return chain && 'partnerChainID' in chain
+}
+
+const isL1Chain = (chain: Chain): chain is L1Network => {
+  return chain && isParentChain(chain) && !isChildChain(chain)
+}
+
+const isL2Chain = (chain: Chain): chain is L2Network => {
+  return chain && isChildChain(chain) && !isOrbitChain(chain)
+}
+
+const isOrbitChain = (chain: Chain): chain is OrbitChain => {
+  return chain && 'isOrbit' in chain && chain.isOrbit
+}
+
+export const getL1Chains = () => {
+  return Object.entries(Networks).reduce((acc, [key, value]) => {
+    if (isL1Chain(value)) {
+      acc[key] = value
+    }
+    return acc
+  }, {} as L1Networks)
+}
+
+export const getL2Chains = () => {
+  return Object.entries(Networks).reduce((acc, [key, value]) => {
+    if (isL2Chain(value)) {
+      acc[key] = value
+    }
+    return acc
+  }, {} as L2Networks)
+}
+
+export const getParentChains = () => {
+  return Object.entries(Networks).reduce((acc, [key, value]) => {
+    if (isParentChain(value)) {
+      acc[key] = value
+    }
+    return acc
+  }, {} as ParentChains)
+}
+
+export const getChildChains = () => {
+  return Object.entries(Networks).reduce((acc, [key, value]) => {
+    if (isChildChain(value)) {
+      acc[key] = value
+    }
+    return acc
+  }, {} as ChildChains)
+}
+
+export const getOrbitChains = () => {
+  return Object.entries(Networks).reduce((acc, [key, value]) => {
+    if (isOrbitChain(value)) {
+      acc[key] = value
+    }
+    return acc
+  }, {} as OrbitChains)
+}
+
+export const getParentForNetwork = (chain: Chain) => {
+  if (!isChildChain(chain)) {
+    return undefined
+  }
+  const parentChain = Networks[chain.partnerChainID]
+  if (!parentChain || !isParentChain(parentChain)) {
+    throw new ArbSdkError(
+      `ParentChain ${chain.partnerChainID} not recognized for Chain ${chain.chainID}.`
+    )
+  }
+  return parentChain
+}
+
+export const getChildrenForNetwork = (chain: Chain) => {
+  if (!isParentChain(chain)) {
+    return undefined
+  }
+  const childChains = getChildChains()
+  const children = Object.values(childChains).filter(
+    childChain => childChain.partnerChainID === chain.chainID
+  )
+  return children
+}
+
+export let l1Networks: L1Networks = getL1Chains()
+export let l2Networks: L2Networks = getL2Chains()
+
+export const getNetwork = async (
   signerOrProviderOrChainID: SignerOrProvider | number,
-  layer: 1 | 2
+  layer?: 1 | 2
```

Comment:
why is this changed to optional now?
i see from the tests that if it's not passed it's an orbit chain?
wouldn't it be better if we type it to `1 | 2 | 'orbit'` or sth like that?